components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -359,6 +359,7 @@ bool CosmeticFiltersJSHandler::ProcessURL(
   }
 
   enabled_1st_party_cf_ =
+      render_frame_->GetWebFrame()->IsCrossOriginToOutermostMainFrame() ||
```

Comment:
Not sure that we need `OutermostMainFrame`, not the nearest one.

[IsCrossOriginToNearestMainFrame](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/frame/frame.h;bpv=1;bpt=1;l=168?q=IsCrossOriginToOutermostMainFrame&gsn=IsCrossOriginToNearestMainFrame&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Dc%252B%252B%3Fpath%3Dthird_party%2Fblink%2Frenderer%2Fcore%2Fframe%2Fframe.h%23XGC7sTPNeUP7R0XDwUYWPXzVTaPyRepfxd1ghrqUM0M)

Reply:
UPD: I believe `IsCrossOriginToOutermostMainFrame` is okay here, because it also cover the case of renced frames
https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/frame/frame.cc;l=212;drc=327a3573ed3d1ee29d433ee215e50947f1b94170;bpv=1;bpt=1